gprofiler/spark/spark_collector.py

```diff
@@ -806,15 +754,18 @@ def _collect_loop(self) -> None:
                         break
 
             if self._spark_sampler is not None:
-                metrics = list(self._spark_sampler.collect())
-                timestamp = datetime.now(tz=timezone.utc)
+                collected = self._spark_sampler.collect()
+                # No need to submit samples that don't actually have a value:
```

Comment:
Where does it actually happen? Or when?

Reply:
If a json field specified in one of the `METRICS` dicts is not present in a json response returned from the yarn/spark server, then a sample can be reported for that metric with value `None`.

Reply:
I see. We have logs for that now though, so we'll need to add if we ever encounter..